File: app/src/main/java/com/example/algorithm/HashTestC.java
package com.example.algorithm;

import java.util.Arrays;
import java.util.HashMap;

public class HashTestC {
    public int solution(String[][] clothes) {
        int answer = 0;
        answer += clothes.length;
        HashMap<String,String> result = new HashMap<>();

        for (int i = 0; i < clothes.length - 1; i++){
            result.put(clothes[i][i],clothes[i].toString());
        }

//        for (int l = 0; l < result.size(); l++){
//            answer = result.get(result.)
//        }

        return answer;
    }
}

File: app/src/main/java/com/example/algorithm/HashTestC.kt
package com.example.algorithm

class HashTestC {

    fun solution(clothes : Array<Array<String>>): Int{
//        clothes.groupBy { it[1] }.forEach {
//            it.value.forEach {
//                it.forEach {
//                    println(it)
//                }
//            }
//        }
        return clothes.groupBy { it[1] }.values.fold(1) { acc, v -> acc * (v.size + 1) }  - 1
    }
}